### PR TITLE
[Oracle SQL] Support parsing of XMLElement and XMLCdata Function

### PR DESCRIPTION
**Pull request by lancelly**
The related issues is: https://github.com/apache/shardingsphere/issues/27569
In this pr, I supported the parsing of basic components of XMLElement and XMLCdata Function of Oracle.
Now, the HR-5-204 & HR-5-208 can be parsed correcly.
The Example SQL is :
`SELECT XMLELEMENT("PurchaseOrder", XMLAttributes(dummy as "pono"), XMLCdata('<!DOCTYPE po_dom_group [ <!ELEMENT po_dom_group(student_name)*> <!ELEMENT po_purch_name (#PCDATA)> <!ATTLIST po_name po_no ID #REQUIRED> <!ATTLIST po_name trust_1 IDREF #IMPLIED> <!ATTLIST po_name trust_2 IDREF #IMPLIED>]>')) "XMLCData" FROM DUAL;`
I refered to the offical doc of Oracle:
https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/XMLELEMENT.html#GUID-DEA75423-00EA-4034-A246-4A774ADC988E
https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/XMLCDATA.html#GUID-FB517A52-6F1A-4D8D-B632-F91EFA606691
截屏2023-08-26 16 25 29
截屏2023-08-26 14 22 26

### CODE REVIEW

**Comment by lancelly** [comment]
It seems that test cases which contains certain special characters are not allowed:
截屏2023-08-26 16 27 15
So for this function, I did not add test cases in the IT.
However, I checked the result statement manually and it seems alright:
截屏2023-08-26 16 29 25

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java [hunk 1]
```diff
@@ -687,11 +688,34 @@ public ASTNode visitXmlFunction(final XmlFunctionContext ctx) {
...
+        if (null != ctx.xmlElementFunction()) {
+            return visit(ctx.xmlElementFunction());
+        }
         FunctionSegment result = new FunctionSegment(ctx.getStart().getStartIndex(), ctx.getStop().getStopIndex(), ctx.specifiedFunctionName.getText(), getOriginalText(ctx));
         result.getParameters().addAll(getExpressions(ctx.exprList()));
         return result;
     }
     
+    @Override
+    public ASTNode visitXmlElementFunction(final OracleStatementParser.XmlElementFunctionContext ctx) {
```
Comment (RaigorJiang) [inline parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java:700]: Please import context class directly.

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java [hunk 2]
```diff
@@ -687,11 +688,34 @@ public ASTNode visitXmlFunction(final XmlFunctionContext ctx) {
...
+        result.getParameters().addAll(expressionSegments);
+        if (null != ctx.xmlAttributes()) {
+            Collection<ExpressionSegment> xmlAttributes = ctx.xmlAttributes().exprWithAlias().stream().map(each -> (ExpressionSegment) visit(each.expr())).collect(Collectors.toList());
+            result.getXmlAttributes().addAll(xmlAttributes);
+        }
+        return result;
+    }
+    
+    @Override
+    public ASTNode visitXmlCdataFunction(final OracleStatementParser.XmlCdataFunctionContext ctx) {
```
Comment (RaigorJiang) [inline parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java:713]: Please import context class directly.

Comment (lancelly) [reply parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java]: done.

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java (hunk 1 above)
Comment (lancelly) [reply parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java]: done.

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**